packages/snap/src/index.ts

```diff
@@ -161,9 +159,28 @@ async function getPrivateKey(bip44Node?: BIP44Node, index: number = 0): Promise<
  * @returns - The address of the optimal feeCurrency, or undefined if the optimal 
  * feeCurrency is Celo. 
  */
-async function getOptimalFeeCurrency(tx: SimpleTransaction): Promise<string | undefined> {
-  // TODO
-  return undefined
+async function getOptimalFeeCurrency(tx: SimpleTransaction, wallet: CeloWallet): Promise<string | undefined> {
+  const gasLimit = (await wallet.estimateGas(tx)).mul(5)
+  const celoBalance = await wallet.getBalance();
+  const addresses = [ //get this dynamically based on network.
+    "0x874069Fa1Eb16D44d622F2e0Ca25eeA172369bC1",
+    "0x10c892A6EC43a53E45D0B916B4b7D383B1b78C0F",
+    "0xE4D517785D091D3c54818832dB6094bcc2744545"
+  ]; 
+  if (gasLimit >= celoBalance) {
+    let promises = [] as any;
```

Comment:
why do we cast this array to type any?

Reply:
yeah if anything  `let promises: Promise<unknown>[]`